src/scripts/main.js

```diff
@@ -1,3 +1,327 @@
 'use strict';
 
 // write code here
+const thead = document.querySelector('thead');
+const tbody = document.querySelector('tbody');
+const tbodyRows = tbody.children;
+const employeeForm = createForm();
+
+thead.addEventListener('click', sortTable);
+tbody.addEventListener('click', sellectRow);
+tbody.addEventListener('dblclick', editCell);
+employeeForm.addEventListener('submit', submitForm);
+
+function submitForm(e) {
+  e.preventDefault();
+
+  const data = new FormData(employeeForm);
+
+  if (checkFields(data)) {
+    return;
+  }
+
+  const dataArr = [
+    data.get('name'),
+    data.get('position'),
+    data.get('office'),
+    data.get('age'),
+    '$' + (+data.get('salary')).toLocaleString(),
+  ];
+
+  const tr = document.createElement('tr');
+
+  tbody.append(tr);
+
+  for (let i = 0; i < dataArr.length; i++) {
+    tr.innerHTML += `<td>${dataArr[i]}</td>`;
+  }
+
+  pushNotification(
+    10,
+    10,
+    'New employee added',
+    'The employee successfully added to the table.',
+    'success',
+  );
+
+  employeeForm.reset();
+}
+
+function checkTableValues(value, headValue) {
+  switch (headValue) {
+    case 'Name':
+      return !(value.length < 4);
+    case 'Age':
+      return !isNaN(+value) && !(+value < 18) && !(+value > 90);
+    case 'Salary':
+      return !isNaN(salaryToNumber(value));
+    case 'Position':
+      return !(value.length === 0);
+    case 'Office':
+      return true;
+  }
+}
+
+function checkFields(formData) {
+  const cheksArr = [];
+
+  if (formData.get('name').length < 4) {
+    pushNotification(
+      150,
+      10,
+      'Wrong name length',
+      'The name must contain 4 characters or more.',
+      'error',
+    );
+
+    cheksArr.push(false);
+  }
+
+  if (+formData.get('age') < 18 || +formData.get('age') > 90) {
+    pushNotification(
+      290,
+      10,
+      'Wrong age',
+      'The age must be between 18 and 90.',
+      'error',
+    );
+
+    cheksArr.push(false);
+  }
+
+  if (!formData.get('position').trim().length) {
+    pushNotification(
+      430,
+      10,
+      'Wrong position',
+      'Please enter a correct position.',
+      'error',
+    );
+
+    cheksArr.push(false);
+  }
+
+  return cheksArr.includes(false);
+}
+
+function editCell(e) {
+  const cell = e.target.closest('td');
+  const curValue = cell.innerText;
+  const theadValue = thead.rows[0].cells[cell.cellIndex].innerText;
+  let cellInput;
+
+  if (!cell || e.target.tagName === 'INPUT') {
+    return;
+  }
+
+  if (cell.cellIndex !== 2) {
+    cellInput = document.createElement('input');
+
+    switch (theadValue) {
+      case 'Name':
+      case 'Position':
+        cellInput.setAttribute('type', 'text');
+        break;
+      case 'Salary':
+      case 'Age':
```

Comment:
Create a variables for these words and use it everywhere

Reply:
Done!